python/flatbuffers/builder.py

```diff
@@ -482,31 +483,40 @@ def Slot(self, slotnum):
         self.current_vtable[slotnum] = self.Offset()
     ## @endcond
 
-    def __Finish(self, rootTable, sizePrefix):
+    def __Finish(self, rootTable, fileIdentifier, sizePrefix):
         """Finish finalizes a buffer, pointing to the given `rootTable`."""
         N.enforce_number(rootTable, N.UOffsetTFlags)
         prepSize = N.UOffsetTFlags.bytewidth
         if sizePrefix:
             prepSize += N.Int32Flags.bytewidth
+        if fileIdentifier:
+            prepSize += FileIdentifierLength
         self.Prep(self.minalign, prepSize)
+
+        if fileIdentifier:
+            b = list(map(ord, fileIdentifier))
```

Comment:
this looks inefficient, creating a new list. Can you somehow roll this into the loop below?

Reply:
done